### PR TITLE
Add edge case tests for Timeboost sequencer

### PR DESCRIPTION
**Pull request by Sneh1999**
### This PR:
Adds edge case tests for timeboost sequencer

### Key places to review:
system_tests/timeboost_sequencer_test.go

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1210554836273498

### CODE REVIEW

**Review by lukeiannucci — COMMENTED** [review]
this looks good to me, thanks for adding the tests. ill wait for more reviews before approving

File: system_tests/timeboost_sequencer_test.go [hunk 1]
```diff
@@ -24,7 +24,6 @@ func createL1AndL2NodeForTimeboost(
 	ctx context.Context,
 	t *testing.T,
 	delayedSequencer bool,
-	blobsEnabled bool,
```
Comment (ImJeremyHe) [inline system_tests/timeboost_sequencer_test.go:27]: Why don't we need this anymore?

File: system_tests/timeboost_sequencer_test.go [hunk 2]
```diff
@@ -209,8 +230,316 @@ func TestEspressoTimeboostSequencer(t *testing.T) {
...
+		}
+		// This check ensures that more blocks were created than the number of inclusion lists
+		if blockNumberAfter-blockNumberBefore <= uint64(numIncls) {
+			t.Fatalf("expected difference between blockNumberAfter and blockNumberBefore should be greater than 0, got: %d", blockNumberAfter-blockNumberBefore)
+		}
+
+		// Initially the round number should be 0 and roundTransactions should contain the transactions from the first inclusion list
+		roundNumber := 0
+		roundsTransactions := make([]*types.Transaction, 0)
+		roundsTransactions = append(roundsTransactions, txnsList[0]...)
```
Comment (ImJeremyHe) [inline system_tests/timeboost_sequencer_test.go:382]: Can't we just initialize `roundsTransactions` with `txnsList[0]`?

**Review by ImJeremyHe — APPROVED** [review]
LGTM. This PR doesn't change any logic and it is always better to have more tests.

File: system_tests/timeboost_sequencer_test.go [hunk 3]
```diff
@@ -209,8 +230,316 @@ func TestEspressoTimeboostSequencer(t *testing.T) {
...
+				txns = append(txns, txn)
+			}
+			txnsList = append(txnsList, txns)
+		}
+		// Generate and send inclusion lists
+		inclusionLists := GenerateInclusionLists(t, users, builder, numIncls, txnsList)
+		SendInclusionLists(t, inclusionLists)
+
+		// Wait for sometime for the block to be produced
+		time.Sleep(time.Second * 120)
```
Comment (lukeiannucci) [inline system_tests/timeboost_sequencer_test.go:363]: does this need to be a 2min delay?

**Review by lukeiannucci — APPROVED** [review]
lgtm

File: system_tests/timeboost_sequencer_test.go (hunk 1 above)
Comment (Sneh1999) [reply system_tests/timeboost_sequencer_test.go:27]: We dont support blobs yet

File: system_tests/timeboost_sequencer_test.go (hunk 3 above)
Comment (Sneh1999) [reply system_tests/timeboost_sequencer_test.go:363]: We dont need it but just to give enough time, we can potentially reduce it. I might as well just merge the PR and optimize these small things later

File: system_tests/timeboost_sequencer_test.go (hunk 2 above)
Comment (Sneh1999) [reply system_tests/timeboost_sequencer_test.go:382]: yes we can, its a test so I will just merge the PR for now
